jax/_src/nn/functions.py

```diff
@@ -186,11 +187,12 @@ def elu(x: Array, alpha: Array = 1.0) -> Array:
   See also:
     :func:`selu`
   """
-  safe_x = jnp.where(x > 0, 0., x)
-  return jnp.where(x > 0, x, alpha * jnp.expm1(safe_x))
+  x = jnp.asarray(x)
+  safe_x = jnp.where(jnp.greater(x, 0), 0., x)
+  return jnp.where(jnp.greater(x, 0), x, alpha * jnp.expm1(safe_x))
```

Comment:
I'd probably go back to using `>` instead of `jnp.greater` – the original code was easier to read